_knitr_hooks.qmd

```diff
@@ -0,0 +1,32 @@
+```{r}
+#| echo: false
+
+
+  # Remove personal directories
+redact <- function(x) {
+  stringr::str_replace_all(
+    x,
+    "/Users/jkeane.*/(PUMS_smol|PUMS cleaned)/",
+    "./data/"
+  )
+}
+
+default_hook_output <- knitr::knit_hooks$get("output")
+
+custom_hook_output <- function(x, options) {
+  x <- redact(x)
+  n <- options$out.lines
```

Comment:
Just observing that `out.lines` is from Danielle's post, but we aren't using it anywhere (yet).

Another thing we should do is truncate long lines, for where we are printing things that don't seem to respond to the `options(width)`. There is one in section 5.2.2 that does `readLines()` on a CSV, and each string then is way longer than 64 chars, and they spill off the page.

Reply:
Yup, totally! I hadn't gotten to that part yet before I needed to leave AMS today 🛬❌🛫🏔️ .

I'm not sure if/when I'll have time to come back to it, so if one of y'all wants to take this branch over that is 100% fine with me.

And yeeeesss about that CSV too